app/src/main/java/org/dhis2/utils/customviews/QRImageViewModel.kt

```diff
@@ -0,0 +1,72 @@
+package org.dhis2.utils.customviews
+
+import android.graphics.Bitmap
+import android.graphics.Color
+import androidx.lifecycle.LiveData
+import androidx.lifecycle.MutableLiveData
+import androidx.lifecycle.ViewModel
+import com.google.zxing.BarcodeFormat
+import com.google.zxing.Writer
+import com.google.zxing.datamatrix.DataMatrixWriter
+import com.google.zxing.oned.Code128Writer
+import com.google.zxing.qrcode.QRCodeWriter
```

Comment:
I'm not sure if it is a good idea to couple a ViewModel with an external librery.
Why don't we create a provider or controller for the feature (qrController) or something like and we decouple these concepts?